napalm_junos/junos.py

```diff
@@ -1325,3 +1325,26 @@ def get_optics(self):
             optics_detail[intf_optic_item[0]] = intf_optics
 
         return optics_detail
+
+    def get_config(self, retrieve='all'):
+        rv = {
+            'startup': '',
+            'running': '',
+            'candidate': ''
+        }
+
+        options = {
+            'format': 'text',
+            'database': 'candidate'
+        }
+
+        if retrieve in ('candidate', 'all'):
+            config = self.device.rpc.get_config(filter_xml=None, options=options)
```

Comment:
The candidate config would make sense if the user is in edit more - otherwise I think it should be empty.
You can introduce a new flag to check that, e.g.: ```pending_changes``` https://github.com/napalm-automation/napalm-iosxr/blob/develop/napalm_iosxr/iosxr.py that must be set in ```_load_candidate```, ```commit_config``` and ```discard_config```